src/assets/scripts/client/aircraft/FlightManagementSystem/WaypointModel.js

```diff
@@ -0,0 +1,44 @@
+import _get from 'lodash/get';
+
+/**
+ *
+ * @class WaypointModel
+ */
+export default class WaypointModel {
+    /**
+     *
+     * @constructor
+     * @param waypointProps {object}
+     */
+    constructor(waypointProps) {
+        this.name = '';
+        this.position = null;
+        this.speedRestriction = -1;
+        this.altitudeRestriction = -1;
+
+        // this.hold = {
+        //     dirTurns: null,
+        //     fixName: null,
+        //     fixPos: null,
+        //     inboundHd: null,
+        //     legLength: null,
+        //     timer: 0
+        // };
```

Comment:
Gotcha. If nothing else, one sucky thing about the way they worked was that the projection thing didn't know how it worked because it depended on game timers to time the legs.